lib/dry/struct/class_interface.rb

```diff
@@ -67,7 +67,7 @@ def constructor_type(type = nil)
 
       def schema
         super_schema = superclass.respond_to?(:schema) ? superclass.schema : {}
-        super_schema.merge(@schema || {})
+        super_schema.merge((defined? @schema) ? @schema : {})
```

Comment:
I am pretty sure @schema is guaranteed to be set @flash-gordon. See https://github.com/dry-rb/dry-struct/blob/6ade17173ea2e7f631007aa7f8b5e85613f0de9f/lib/dry/struct/class_interface.rb#L17